src/NuGet.Core/NuGet.Protocol/Plugins/Connection.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Versioning;
+
+namespace NuGet.Protocol.Plugins
+{
+    /// <summary>
+    /// Represents a bidirectional channel between a NuGet client and a plugin.
+    /// </summary>
+    public sealed class Connection : IConnection
```

Comment:
In general, avoid common names for public types. I'd suggest `PluginConnection` for this one.